components/channel_layout/center_channel/center_channel.tsx

```diff
@@ -58,6 +69,27 @@ export default class CenterChannel extends React.PureComponent<Props, State> {
         return {lastReturnTo: nextProps.location.pathname};
     }
 
+    async componentDidMount() {
+        const {actions, showNextSteps, showNextStepsTips, isOnboardingHidden, currentUserId} = this.props;
+        await actions.getProfiles();
+
+        // We call getUser to make sure we have the current user's profile into the profiles state
+        await actions.getUser(currentUserId);
+        if ((showNextSteps || showNextStepsTips) && !isOnboardingHidden) {
+            actions.setShowNextStepsView(true);
+        }
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        const {location, showNextStepsEphemeral, actions, currentUserId} = this.props;
+        if (location.pathname !== prevProps.location.pathname && showNextStepsEphemeral) {
+            actions.setShowNextStepsView(false);
+        }
+
+        // We call getUser to make sure we have the current user's profile into the profiles state
+        actions.getUser(currentUserId);
```

Comment:
This will be called a LOT of times if we do it everytime the component refreshes. Are we just doing it in case it changes? If so, it might make more sense to have it pulled down by the action that changes it instead of having it update everytime the component does.

Reply:
Yes. It's possible that `getProfiles` gets called else where. In fact, if it it'll remove the current user from the profiles. See here https://github.com/mattermost/mattermost-webapp/blob/42ac676894524deb94743e3243653f1c9035b201/packages/mattermost-redux/src/actions/users.ts#L304

Reply:
Why do we need to get the user for every update here?

Reply:
It’s the only way to ensure the current user profile is part of the profiles fetched by ‘getProfiles’

Reply:
Sure, taking a step back here - why do we need to fetch profiles on the center channel ? Apologies for insisting here, but just want to make the correlation with onboarding and why is something we need now

Reply:
We probably don’t need to fetch profiles. It’s something I took from the previous implementation. But 4/5 we need to getUser

Reply:
Actually, I have remembered, the reason I had put it there was to check for the user creation date from the profile. We're not doing that anymore. I'll take it out :)

Reply:
With the `getProfiles`, since it was used in the previous implementation, I am not sure why they call it, so it's safe to leave it there.

Reply:
Makes sense! Thanks for double checking 👍

Reply:
@devinbinnie sorry for the confusion with the call to `getUser`. I have taken it out.